File: src/main/java/net/ehicks/bts/beans/Severity.java
//package net.ehicks.bts.beans;
//
//import net.ehicks.bts.ISelectTagSupport;
//import net.ehicks.eoi.EOI;
//
//import javax.persistence.*;
//import java.io.Serializable;
//import java.util.Arrays;
//import java.util.List;
//
//@Entity
//@Table(name = "severities")
//public class Severity implements Serializable, ISelectTagSupport
//{
//    @Id
//    @GeneratedValue(strategy=GenerationType.AUTO)
//    private Long id;
//
//    @Column(nullable = false, unique = true)
//    private String name = "";
//
//    @Override
//    public boolean equals(Object obj)
//    {
//        if (!(obj instanceof Severity)) return false;
//        Severity that = (Severity) obj;
//        return this.id.equals(that.getId());
//    }
//
//    @Override
//    public int hashCode()
//    {
//        return 17 * 37 * id.intValue();
//    }
//
//    public String toString()
//    {
//        return this.getClass().getSimpleName() + ":" + id.toString();
//    }
//
//    @Override
//    public String getValue()
//    {
//        return id.toString();
//    }
//
//    @Override
//    public String getText()
//    {
//        return name;
//    }
//
//    public static List<Severity> getAll()
//    {
//        return EOI.executeQuery("select * from severities");
//    }
//
//    public static Severity getById(Long id)
//    {
//        return EOI.executeQueryOneResult("select * from severities where id=?", Arrays.asList(id));
//    }
//
//    // -------- Getters / Setters ----------
//
//    public Long getId()
//    {
//        return id;
//    }
//
//    public void setId(Long id)
//    {
//        this.id = id;
//    }
//
//    public String getName()
//    {
//        return name;
//    }
//
//    public void setName(String name)
//    {
//        this.name = name;
//    }
//}

File: src/main/java/net/ehicks/bts/beans/Severity.kt
package net.ehicks.bts.beans

import net.ehicks.bts.ISelectTagSupport
import net.ehicks.bts.Named
import org.hibernate.envers.Audited
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.io.Serializable
import javax.persistence.*

@Entity
@Audited
data class Severity(
        @Id @GeneratedValue(strategy = GenerationType.AUTO)
        private val id: Long = 0,
        @Column(unique = true)
        private val name: String = ""
) : Serializable, ISelectTagSupport, Named {
    override fun getText(): String {
        return name
    }

    override fun getValue(): String {
        return id.toString()
    }

    override fun getId(): Long { return id}
    override fun getName(): String { return name}
}

@Repository
interface SeverityRepository : JpaRepository<Severity, Long> {
    fun findByName(name: String): Severity
    fun findByIdIn(ids: List<Long>): Set<Severity>
}